olp-cpp-sdk-core/include/olp/core/client/BackdownStrategy.h

```diff
@@ -64,5 +64,54 @@ struct CORE_API ExponentialBackdownStrategy {
   }
 };
 
+/**
+ * @brief Computes wait time for the next retry attempt via
+ * the exponential backoff with the added jitter.
+ *
+ * This backoff strategy is based on the Equal Jitter approach. See
+ * https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+ *
+ * The actual formula can be described in the following way:
+ * @code{.unparsed}
+ * temp = min(cap, base * 2 ** attempt)
+ * sleep = temp / 2 + random_between(0, temp / 2)
+ * @endcode
+ */
+struct CORE_API EqualJitterBackdownStrategy {
+ public:
+  /**
+   * @brief Creates a EqualJitterBackdownStrategy instance.
+   *
+   * @param min_cap The minimum cap (ms) used in the wait time formula.
+   */
+  explicit EqualJitterBackdownStrategy(std::chrono::milliseconds cap)
```

Comment:
I would not do it in the constructor for the simple reason as the user does not know what the cap should be and he will be tempted to set the max_wait_time here which is to be considered only after all the retries are done. The cap itself is in my opinion variable and therefore more suitable to be added to the operator() as it will be adapted by the OlpClient based on the current retry and what time is left from the previous retries to the max_wait_time.

Reply:
We can't change the operator() signature since it is a breaking change.